index.html

```diff
@@ -3852,51 +3973,47 @@ <h3>Discovery Examples</h3>
         Using the {{Symbol/asyncIterator}} provided by the Discovery object, we can iterate asynchronously over the results and perform operations with the obtained {{ThingDescription}} objects.
       </p>
       <pre class="example" title="Fetch the Thing Description of a Thing">
-        let discovery = new ThingDiscovery({ method: "direct" });
-        for await (let td of discovery) {
-          console.log("Found Thing Description for " + td.title);
-        }
+        let url = "https://mythings.com/thing1";
+        let td = await WOT.requestThingDescription(url);
+        console.log("Found Thing Description for " + td.title);
       </pre>
       <p>
         The next example finds {{ThingDescription}} objects of <a>Thing</a>s listed in a <a>Thing Directory</a> service. We set a timeout for safety.
       </p>
       <pre class="example" title="Discover Things via directory">
-        let discoveryFilter = {
-          method: "directory", // default value, no need to specify
-          url: "http://directory.wotservice.org"
-        };
-        let discovery = new ThingDiscovery(discoveryFilter);
+        let discovery = WOT.exploreDirectory("http://directory.wotservice.org");
         setTimeout( () => {
             discovery.stop();
             console.log("Discovery stopped after timeout.");
           },
           3000);
-        do {
-          let td = await discovery.next();
+        for await (const td of discovery) {
           console.log("Found Thing Description for " + td.title);
           let thing = new ConsumedThing(td);
           console.log("Thing name: " + thing.getThingDescription().title);
-        } while (!discovery.done);
+        };
         if (discovery.error) {
           console.log("Discovery stopped because of an error: " + error.message);
         }
       </pre>
-      <!--p>
-        The next example is for an open-ended multicast discovery, which likely won't complete soon (depending on the underlying protocol), so stopping it with a timeout is a good idea. It will likely deliver results one by one.
+      <p>
+        The next example is for a generic discovery, by any means provisioned
+        to the WOT runtime, including local Things, if any is available.
       </p>
       <pre class="example" title="Discover Things in a network">
-        let discovery = new ThingDiscovery({ method: "multicast" });
+        let discovery = WOT.discover();
```

Comment:
With the new API shape, I suppose we need an `await` here, right?

```suggestion
        const discovery = await WOT.discover();
```